Dockerfile

```diff
@@ -1,15 +1,16 @@
-FROM alpine:3.11.3 as base
-RUN apk add git python3 gcc g++ linux-headers make
+FROM alpine:3.14.0 as base
+RUN apk add git gcc g++ linux-headers make
 WORKDIR /usr/src/app
 
 FROM base as fetch
 ARG NODE_VERSION=v16.4.2
+RUN if [ "${NODE_VERSION:0:3}" = v12 ]; then apk add python2; else apk add python3; fi
```

Comment:
I found that the `./configure` script for Node.js v12 would complain if Python 3 was installed instead of Python 2:
```
#9 [configure 2/2] RUN ./configure --fully-static --enable-static
#9 sha256:81658adf319913f0b885cb9baa3ebc9c6a54c638dac36252a7a8879fd6591960
#9 0.324 ./configure: exec: line 4: python: not found
#9 ERROR: executor failed running [/bin/sh -c ./configure --fully-static --enable-static]: exit code: 127
```

And similarly, the `./configure` script for Node.js v14 or later would complain if Python 2 was installed instead of Python 3:
```
#9 [configure 2/2] RUN ./configure --fully-static --enable-static
#9 sha256:08c919df891205257ab5d1d0478436254ff8c12e97320272c49f32c31a3abd0b
#9 0.991 Node.js configure: Found Python 2.7.18...
#9 0.991 Please use python3.9 or python3.8 or python3.7 or python3.6.
#9 ERROR: executor failed running [/bin/sh -c ./configure --fully-static --enable-static]: exit code: 1
```

I.e., if Node v12, we need Python v2, otherwise we need Python v3.

* The package name for Python v2 is `python` in alpine v3.11.3, but `python2` in alpine v3.14.0.
* The package name for Python v3 is `python3` in both versions of alpine.

Reply:
Good catch